src/Codeception/Lib/GroupManager.php

```diff
@@ -118,6 +118,12 @@ public function groupsForTest(\PHPUnit\Framework\Test $test)
                 if (strpos($filename . ':' . $test->getName(false), $testPattern) === 0) {
                     $groups[] = $group;
                 }
+                // TODO: consider mb_strtolower per https://stackoverflow.com/a/5473569
+                if (method_exists($test, 'getMetadata')
+                    && strcasecmp($filename . ':' . $test->getMetadata()->getFeature(), $testPattern) === 0
```

Comment:
Since current solution is focused on Gherkin feature, before using `$test->getMetadata()->getFeature()`, you should verify that the current Test object is a [Gherkin Test](https://github.com/Codeception/Codeception/blob/cc739e8a049309842e88d8f5389f47826549ca7f/src/Codeception/Test/Gherkin.php)

Reply:
I can do that. Are there any other types of tests that might benefit from this comparison though?